docs/BacwardCompatibility.md

```diff
@@ -0,0 +1,239 @@
+releases v1.4.x and v1.3.x, on another hand it might work with releases v1.2.x, but it is not guaranteed 
+
+## Fybrik components or features that can influence compatability 
+
+*Note:* the list is not closed, and can be extended
+
+*Note:* some components, listed below, are external integration points, changes in them impacts not only on the backward 
+compatibility, but on external development process. We emphasized these components with the label `integration point`.  
+
+- Fybrik control plane [Kubernetes Custom Resources](#kubernetes-custom-resource-versions-and-upgrade-plan) 
+`FybrikApplication` is a Fybrik entry point and `FybrikModule` is a definition for externally developed modules, 
+therefore these CRDs are integration points. 
+- [Connectors](#connectors): are Open API services that the Fybrik control plane uses to connect to external systems. 
+These connector services are deployed alongside the Fybrik control plane, and as we can see from their name, all of them 
+are external integration points
+  - Data Catalog connector
+  - Policy Manager connector
+  - Credential Management connector
+- [Default Taxonomy](#default-taxonomy). Terms and values defined by taxonomy are part of Connectors APIs and CRDs 
+validations, therefore we can see taxonomy as part of external connectivity and an integration point.  
+- [Default Policies](#default-policies)
+  - [Data Access Policies](#default-data-access-policies)
+  - [IT Configuration Policies](#default-it-config-policies)
+- [Fybrik Logs format](#fybrik-logs-format). If Fybrik logs are used by external analytic tools, they will be part of 
+external integration point too.
+
+## Kubernetes Custom Resource Versions and Upgrade Plan
+
+Current Fybrik [Custom Resource Definitions](https://github.com/fybrik/fybrik/tree/master/manager/apis/app/v1beta1) 
+
+### Versions
+Fybrik for its Kubernetes API objects (CRDs) uses API versioning like one used by
+[Kubernetes](https://kubernetes.io/docs/reference/using-api/#api-versioning).
+- Alpha
+    - The version names contain alpha (for example, v1alpha1).
+    - The software is not stable may contain bugs.
+    - The support for a feature may be dropped at any time without notice.
+    - The API may change in incompatible ways in a later software release without notice.
+- Beta
+    - The version names contain beta (for example, v2beta3).
+    - The software is well tested. Enabling a feature is considered safe.
+    - The support for a feature will not be dropped, though the details may change.
+    - The schema and/or semantics of objects may change in incompatible ways in a subsequent beta or stable release.
+      When this happens, migration instructions are provided. Schema changes may require deleting, editing, and re-creating
+      API objects. The editing process may not be straightforward. The migration may require downtime for applications that
+      rely on the feature.
```

Comment:
Question: let's say I am changing a schema in a backwards-compatible way, e.g. making a required field optional, removing an optional field, adding a new optional field while there is a default value for an empty field, etc. Should the CRD version be changed and how?

Reply:
You don't have to.

> "additive optional fields don't require version bumps"

see discussion https://kubernetes.slack.com/archives/C0EG7JC6T/p1662443350841479